website/static/js/pages/dashboard-page.js

```diff
@@ -40,6 +41,104 @@ request.fail(function(xhr, textStatus, error) {
+            mergeMessage = 'Would you like to add \<b>' + email[0].address + '\</b> to your account?';
+            confirmMessage = '\<b>' + email[0].address + '\</b> has been added into your account.';
+            nopeMessage = 'You have chosen not to add \<b>' + email[0].address + '\</b> to your account.' +
+             'If you change your mind, visit the \<a href="/settings/account/">user settings page</a>.';
+        }
+        var failMessage = 'There was a problem. Please contact <a href="mailto: support@osf.io">support@osf.io</a> ' +
+        'if the problem persists.';
+
+        bootbox.dialog({
+            title: title,
+            message: mergeMessage,
+            onEscape: function() {},
+            closeButton: true,
+            buttons: {
+                confirm: {
+                    label: 'Add email',
+                    className: 'btn-success',
+                    callback: function () {
+                        $osf.putJSON(
+                            confirmedEmailURL,
+                            email[0]
+                        ).done(function () {
+                            $osf.growl('Success', confirmMessage, 'success', 3000);
+                        }).fail(function (xhr, textStatus, error) {
+                            Raven.captureMessage('Could not set user timezone or locale', {
+                                url: url,
+                                textStatus: textStatus,
+                                error: error
+                            });
+                            $osf.growl('Error',
+                                failMessage,
+                                'danger'
+                            );
+                        });
+                        confirm_emails(emails);
+                    }
+                },
+                cancel: {
+                    label: 'Cancel',
+                    className: 'btn-default',
+                    callback: function () {
+                        $osf.putJSON(
+                            removeConfirmedEmailURL,
+                            email[0]
+                        ).done(function () {
+                            $osf.growl('Warning', nopeMessage, 'warning', 8000);
+                        }).fail(function (xhr, textStatus, error) {
+                            Raven.captureMessage('Could not set user timezone or locale', {
+                                url: url,
+                                textStatus: textStatus,
+                                error: error
+                            });
+                            $osf.growl('Error',
+                                failMessage,
+                                'danger'
+                            );
+                        });
+                        confirm_emails(emails);
+                    }
+                }
+            }
+        });
+
+    }
+}
+
+request = $.getJSON(confirmedEmailURL, function(response) {
```

Comment:
Is there a way to avoid doing this every time, like some sort of flag that would indicate whether or not it's necessary.

Also, if not, why not do something like this instead of assigning an otherwise unused variable:
```js
$.getJSON(confirmedEmailURL).done(
    function(){...}
).fail(
    function(){...}
);
```